cmd/rosetta-dispatcher-server/main.go

```diff
@@ -0,0 +1,297 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	url2 "net/url"
+	"os"
+	"os/signal"
+	"sort"
+	"time"
+
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/echo/v4/middleware"
+	"github.com/optakt/flow-dps/rosetta/identifier"
+	"github.com/rs/zerolog"
+	"github.com/spf13/pflag"
+	"github.com/ziflex/lecho/v2"
+)
+
+const (
+	success = 0
+	failure = 1
+)
+
+type Spork struct {
+	First      uint64
+	Last       uint64
+	ProxyTaget *middleware.ProxyTarget
+}
+
+type SporkList []Spork
+
+func NewSporkList(addresses []string, firsts []int64, lasts []int64) (SporkList, error) {
+	// TODO: Add tests for validating sporks configuration
+
+	// Check past sporks configuration
+	if len(addresses) != len(firsts) || len(addresses) != len(lasts) {
+		return nil, fmt.Errorf("data length mismatch")
+	}
+
+	if len(addresses) < 2 {
+		return nil, fmt.Errorf("at least two sporks must be provided")
+	}
+
+	sporksTable := make(SporkList, len(addresses))
+
+	for i, sporkAddress := range addresses {
+		sporksTable[i].First = uint64(firsts[i])
+		sporksTable[i].Last = uint64(lasts[i])
+
+		url, err := url2.Parse(sporkAddress)
+		if err != nil {
+			return nil, fmt.Errorf("spork address %d is invalid: %w", i, err)
+		}
+		if sporksTable[i].First >= sporksTable[i].Last {
+			return nil, fmt.Errorf("spork %d last height is not greater than first", i)
+		}
+
+		sporksTable[i].ProxyTaget = &middleware.ProxyTarget{URL: url}
+	}
+
+	sort.Sort(sporksTable)
+
+	for i := 0; i < len(sporksTable)-1; i++ {
+		if sporksTable[i].Last != sporksTable[i+1].First-1 {
+			return nil, fmt.Errorf("gap between sporks boundaries: %d and %d", sporksTable[i].Last, sporksTable[i+1].First)
+		}
+	}
```

Comment:
I feel like this could be done differently. Also it might panic here, since when we check the last element of the slice, we compare it to the next one which does not exist.

For example with a slice of length 5, `i` would be equal to `4` on the last iteration, and when running the `if` statement we would effectively do: `if sporksTable[4].Last != sporksTable[5].First-1` which would panic.